src/Metadata/Util/ArrayTrait.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Metadata\Util;
+
+trait ArrayTrait
```

Comment:
this is not internal? can you tell me where its used? We need to cover backward compatibility on the old one and/or probably fix this not being internal but I'm not sure.